internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -0,0 +1,313 @@
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachines/finalizers,verbs=*
+// +kubebuilder:rbac:groups=*,resources=persistentvolumeclaims,verbs=*
+// +kubebuilder:rbac:groups=*,resources=pods,verbs=create
+// +kubebuilder:rbac:groups=*,resources=secrets,verbs=*
+
+const (
+	operandName      = "tekton-tasks"
+	operandComponent = common.AppComponentTektonTasks
+
+	cleanVMTaskName              = "cleanup-vm"
+	copyTemplateTaskName         = "copy-template"
+	modifyDataObjectTaskName     = "modify-data-object"
+	createVMFromTemplateTaskName = "create-vm-from-template"
+	diskVirtCustomizeTaskName    = "disk-virt-customize"
+	diskVirtSysprepTaskName      = "disk-virt-sysprep"
+	modifyTemplateTaskName       = "modify-vm-template"
+	waitForVMITaskName           = "wait-for-vmi-status"
+	createVMFromManifestTaskName = "create-vm-from-manifest"
+	generateSSHKeysTaskName      = "generate-ssh-keys"
+	executeInVMTaskName          = "execute-in-vm"
+	modifyWindowsVMIsoFileName   = "modify-windows-iso-file"
+)
+
+var requiredCRDs = []string{"tasks.tekton.dev"}
+
+var AllowedTasks = map[string]func() string{
+	createVMFromManifestTaskName: environment.GetCreateVMImage,
+	cleanVMTaskName:              environment.GetCleanupVMImage,
+	copyTemplateTaskName:         environment.GetCopyTemplateImage,
+	modifyDataObjectTaskName:     environment.GetModifyDataObjectImage,
+	createVMFromTemplateTaskName: environment.GetCreateVMImage,
+	diskVirtCustomizeTaskName:    environment.GetDiskVirtCustomizeImage,
+	diskVirtSysprepTaskName:      environment.GetDiskVirtSysprepImage,
+	modifyTemplateTaskName:       environment.GetModifyVMTemplateImage,
+	waitForVMITaskName:           environment.GetWaitForVMIStatusImage,
+	generateSSHKeysTaskName:      environment.GetSSHKeysStatusImage,
+	executeInVMTaskName:          environment.GetCleanupVMImage,
+	modifyWindowsVMIsoFileName:   environment.GetCleanupVMImage,
+}
+
+func init() {
+	utilruntime.Must(pipeline.AddToScheme(common.Scheme))
+}
+
+type tektonTasks struct {
+	tasks           []pipeline.Task
+	serviceAccounts []v1.ServiceAccount
+	roleBindings    []rbac.RoleBinding
+	clusterRoles    []rbac.ClusterRole
+}
+
+var _ operands.Operand = &tektonTasks{}
+
+func New(bundle *tektonbundle.Bundle) *tektonTasks {
+	tt := &tektonTasks{
+		tasks:           bundle.Tasks,
+		serviceAccounts: bundle.ServiceAccounts,
+		roleBindings:    bundle.RoleBindings,
+		clusterRoles:    bundle.ClusterRoles,
+	}
+
+	tt.filterUnusedObjects()
+
+	return tt
+}
+
+func (t *tektonTasks) Name() string {
+	return operandName
+}
+
+func (t *tektonTasks) WatchClusterTypes() []client.Object {
+	return []client.Object{
+		&rbac.ClusterRole{},
+		&pipeline.ClusterTask{},
+		&rbac.RoleBinding{},
+		&v1.ServiceAccount{},
+	}
+}
+
+func (t *tektonTasks) WatchTypes() []client.Object {
+	return nil
+}
+
+func (t *tektonTasks) RequiredCrds() []string {
+	return requiredCRDs
+}
+
+func (t *tektonTasks) filterUnusedObjects() {
```

Comment:
`code style` - This does not need to be a method. It can the directly inlined in `New()` function, and the slices can be filtered before the struct is created.

Reply:
Thanks! It solved an issue that there was no a way to test resource creation. This `filterUnusedObjects` modified only private struct variables and not the bundle (that used in unit tests).